### DISCUSSION TITLE
Updating numeric entry fields from API call

### DISCUSSION THREAD

**Discussion by ttelik** [Budibase/budibase#7597] [Help]
Hi Everyone,

I am trying to use an API call to update a number field in a form. The way I implemented this is a button press triggers an API call, then the returned value is assigned to a state variable on the page. 

I use the state variable to set the default value for the number field. This has a strange behavior - a zero returned from the API call doesn't update the cell. Also, neither do small numbers (i.e. 0.001). I would also like to be able to fetch new values with repeated button presses, but this doesn't work if the value is used to set the "default value".

What's the best way to implement this kind of functionality? 

Just to elaborate a little further, from the user perspective the workflow is:
- The user presses a button to trigger the API call.
- Data is loaded into a form that is formatted in a sensible way (I don't want to use a data provider + table view)
- Once the user is happy with the data they press save, which saves to a Table.
- The user will repeat this sequence several times starting at the top, so we need to refresh the values as they change.

Thanks in advance.

**Comment by melohagan** [comment]
Hi @ttelik 

The default value only sets on load of the page. 

If you want to update a form field you will need to use the **Update Field Value** action: https://docs.budibase.com/docs/actions#update-field-value
